mods/content/psionics/system/psionics/complexus/complexus_process.dm

```diff
@@ -98,13 +118,22 @@
 			if(stamina > 10)
 				stamina = max(0, stamina - rand(15,20))
 				to_chat(owner, SPAN_DANGER("You feel your psi-power leeched away by \the [psi_leech]..."))
-			else
-				stamina++
-		else if(stamina < max_stamina)
+//			else
+//				stamina++
+	/*	else if(stamina < max_stamina) 					OFSTATION EDIT BEGIN
 			if(owner.stat == CONSCIOUS)
 				stamina = min(max_stamina, stamina + rand(1,3))
 			else if(owner.stat == UNCONSCIOUS)
-				stamina = min(max_stamina, stamina + rand(3,5))
+				stamina = min(max_stamina, stamina + rand(3,5))*/
+													//OFSTATION EDIT BEGIN
+		else if(stamina < max_stamina)
+			stamina = min(max_stamina, stamina + traumastam_regen + cyberstam_regen)
+			if(stamina <= 0)
+				stamina = 0
+				suppressed = TRUE
+				update_hud = TRUE
+			adjust_psi_gen()
```

Comment:
`adjust_psi_gen()` should process every time, regardless of any condition, considering it should be constantly degrading or increasing. Any conditional checks regarding stamina should be done within it.